pkg/disk/disk.go

```diff
@@ -0,0 +1,354 @@
+	if err != nil {
+		return nil, grpcstatus.Error(grpccodes.Internal, errors.Wrap(err, "failed to get SPDK client").Error())
+	}
+	defer spdkClient.Close()
+
+	diskPath := getDiskPath(req.DiskPath)
+
+	bdevName, err := spdkClient.BdevAioCreate(diskPath, req.DiskName, blockSize)
+	if err != nil {
+		resp, parseErr := parseErrorMessage(err.Error())
+		if parseErr != nil || !isFileExists(resp.Message) {
+			log.WithError(err).Error("Failed to create AIO bdev")
+			return nil, grpcstatus.Error(grpccodes.Internal, errors.Wrap(err, "failed to create AIO bdev").Error())
+		}
+	}
+
+	lvstoreName := bdevName
+	lvstoreInfos, err := spdkClient.BdevLvolGetLvstore("", "")
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get lvstore with name %v", lvstoreName)
+		return nil, grpcstatus.Error(grpccodes.Internal, errors.Wrapf(err, "failed to get lvstore with name %v", lvstoreName).Error())
+	}
+
+	uuid := ""
+	switch len(lvstoreInfos) {
+	case 0:
+		uuid, err = spdkClient.BdevLvolCreateLvstore(lvstoreName, req.DiskName, defaultClusterSize)
+		if err != nil {
+			log.WithError(err).Error("Failed to create lvstore")
+			return nil, grpcstatus.Error(grpccodes.Internal, errors.Wrapf(err, "failed to create lvstore").Error())
+		}
+	case 1:
+		log.Infof("Found an existing lvstore %v", lvstoreInfos[0].Name)
+		lvstoreInfo := &lvstoreInfos[0]
+		if lvstoreInfo.Name != lvstoreName {
+			log.Infof("Renaming the existing lvstore %v to %v", lvstoreInfo.Name, lvstoreName)
+			renamed, err := spdkClient.BdevLvolRenameLvstore(lvstoreInfo.Name, lvstoreName)
+			if err != nil {
+				log.WithError(err).Errorf("Failed to rename lvstore from %v to %v", lvstoreInfo.Name, lvstoreName)
+				return nil, grpcstatus.Error(grpccodes.Internal, errors.Wrapf(err, "failed to rename lvstore from %v to %v", lvstoreInfo.Name, lvstoreName).Error())
+			}
+			if !renamed {
+				log.Errorf("Failed to rename lvstore from %v to %v", lvstoreInfo.Name, lvstoreName)
+				return nil, grpcstatus.Errorf(grpccodes.Internal, "failed to rename lvstore from %v to %v", lvstoreInfo.Name, lvstoreName)
+			}
+		}
+		uuid = lvstoreInfo.UUID
+	default:
+		log.Error("Found more than one lvstore")
+		return nil, grpcstatus.Error(grpccodes.Internal, "found more than one lvstore")
+	}
+
+	lvstoreInfo, err := s.bdevLvolGetLvstore(spdkClient, "", uuid)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get lvstore with name %v and UUID %v", req.DiskName, uuid)
+		return nil, grpcstatus.Error(grpccodes.Internal, errors.Wrapf(err, "failed to get lvstore with name %v and UUID %v", req.DiskName, uuid).Error())
+	}
+
+	log.Info("Created disk")
+
+	// A disk does not have a fsid, so we use the device number as the disk ID
+	diskID, err := getDeviceNumber(diskPath)
+	if err != nil {
+		log.WithError(err).Error("Failed to get disk ID")
+		return nil, grpcstatus.Error(grpccodes.Internal, errors.Wrapf(err, "failed to get disk ID").Error())
+	}
+
+	return &rpc.Disk{
+		Id:          diskID,
```

Comment:
Actually, I'm not comfortable with the major-minor number as an ID. I'm thinking if we can let ID is same as UUID.
`ID` is `fsid` for filesystem-type disk, so it is very unique. For a block-type disk, UUID might be better. WDYT?

Reply:
1. We can store the disk name (which is provided by users and is the lvs-name for SPDK) in a new field `name`
2. As for this field `ID`, it can be the major-minor for SPDK/block disk and fsid for ds disk. I don't think we need to make this the same as `UUID` as we already have that field.

Reply:
I see. Let's use major-minor. How about it?